R/add_creator_id.R

```diff
@@ -0,0 +1,67 @@
+#' add_creator_id
+#'
+#' This function allows you to add an ORCID or reference ID to a creator in EML.
+#' @param eml EML script to modify
+#' @param orcid ORCID in the format 'https://orcid.org/WWWW-XXXX-YYYY-ZZZZ'
+#' @param id reference ID. Character string to set reference ID for creators with additional roles (i.e. metadataProvider, etc.)
+#' @param surname creator surname (last name), defaults to first creator if not specified. Not case-sensitive.
+#' @keywords eml creator orcid id
+#' @details All prameters other than the EML are optional, but since the point of the function is to modify either the orcid, ref id, or both, you better specify at least one. Requires the crayon package.
+#' @export
+#' @examples
+#' add_creator_id(eml2, orcid = "https://orcid.org/WWWW-XXXX-YYYY-ZZZZ")
+#'
+
+#function for adding orcid or reference id to creator
+add_creator_id <- function(eml,
+                           orcid = NULL,
+                           id = NULL,
+                           surname = NULL) { #not sensitive to capitalization
+    library(crayon)
+
+    #grab creators and save to new variable
+    creatorList <- eml@dataset@creator@.Data
+
+    #determine surname position to access correct creator
+    #if none are specified, the first creator will be modified
+    if(is.null(surname == TRUE)){
+        cat(green("Since surname was not specified, the first creator entry will be modified. "))
+        pos <- 1
+    } else {
+        #make vector of creator surnames
+        surNames <- rep(NA, times = length(creatorList))
```

Comment:
If you're referencing arguments in a function in order they're usually not included.

Reply:
Not sure if this is the best etiquette. Probably a lot better to not assume users want first creator. Requiring `surname` and `givenName` would be best practice... and both may be necessary as two people could have the same `surname` or `givenName`

Reply:
Also the code is broken if no surname is entered.

Reply:
In a lot of cases, there is only one creator, which is why I chose to default to the first entry (but print a message so people are aware). As it stands, no surname works ok.

Two creators with the same surname is plausible, but I'll just add that to the "to-do" list for now